### PR TITLE
Add AWS SSM functionality

### LINKED ISSUE #26: AWS SSM install broken on ubuntu
AWS SSM disregards shebang and it will run the script using `/bin/dash` a minimalistic POSIX shell.

The `dash` support has been broken lately:
```
# dash falcon-linux-deploy.sh 
falcon-linux-deploy.sh: 89: falcon-linux-deploy.sh: Syntax error: "(" unexpected (expecting "}")
```

seems that `( )` bellow is bashism
```
    sensor_versions=( )
    for i in $sensor_update_versions; do
        sensor_versions+=("$i")
    done
```

### PR DESCRIPTION
**Pull request by redhatrises**
- Fixes #26

### CODE REVIEW

File: falcon-linux-deploy.sh [hunk 1]
```diff
@@ -381,6 +449,8 @@ cs_falcon_cid=$(
     # shellcheck disable=SC2154
     if [ -n "$FALCON_CID" ]; then
         echo "$FALCON_CID"
+    else
+        aws_ssm_parameter "FALCON_CID" | json_value Value 1
```
Comment (redhatrises) [inline falcon-linux-deploy.sh:453]: @isimluk in the original, this is `FalconCID`. I assume that we can switch this to `FALCON_CID`.

Comment (isimluk) [reply falcon-linux-deploy.sh:454]: I think we will also have to fetch client_id and client_secret from SSM.

Comment (isimluk) [reply falcon-linux-deploy.sh:454]: This is what I mean: https://github.com/CrowdStrike/Cloud-AWS/blob/main/Agent-Install-Examples/bash/API-download/install.sh#L364-L378

Comment (redhatrises) [reply falcon-linux-deploy.sh:454]: Right. That's what I had missed earlier which I thought I did an accurate search, but clearly, as evidenced by this pr, that no... no I did not. /me--

Comment (redhatrises) [reply falcon-linux-deploy.sh:464]: Okay... it is fixed.

File: falcon-linux-deploy.sh [hunk 2]
```diff
@@ -361,9 +429,19 @@ cs_os_version=$(
     fi
 )
 
+aws_instance=$(
+    curl -s --connect-timeout 5 http://169.254.169.254/latest/dynamic/instance-identity/
```
Comment (isimluk) [inline falcon-linux-deploy.sh:433]: I don't like the idea of calling home to AWS any time we install.

There has to be a way to figure out we are on EC2 without the networking.

Comment (redhatrises) [reply falcon-linux-deploy.sh]: Okay... I think I have a solution. See what you think.

**Comment by redhatrises** [comment]
dash shell fixes are in #40

File: falcon-linux-deploy.sh [hunk 3]
```diff
@@ -361,9 +429,28 @@ cs_os_version=$(
...
+    else
+        curl_output="$(curl -s --connect-timeout 5 http://169.254.169.254/latest/dynamic/instance-identity/)"
+        if [ -n "$curl_output" ] && ! echo "$curl_output" | grep 'Not Found'; then
+            echo true
+        fi
+    fi
+)
+
+aws_my_region=$(
+    curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone | sed s/.$//
```
[the remark this answers is not in the document]
Comment (redhatrises) [reply falcon-linux-deploy.sh]: Good point... moved that into the `aws_ssm_parameter` function.

**Comment by isimluk** [comment]
Would you be willing to rebase on top of the main branch to show dash shellchecker fixes?

**Review by isimluk — APPROVED** [review]
👍 👌

**Pull request merged**